src/fileTransformers/mysqlDataProvider.ts

```diff
@@ -0,0 +1,26 @@
+import { FileTransformer } from '~/src/fileTransformer/fileTransformer'
+
+export const mysqlDataProvider: FileTransformer = (params) => {
+  const { file, parameters } = params
+
+  if (parameters.datasourceProvider !== 'mysql') {
+    return file.content
+  }
+
+  const replaceTextConfigList = [
+    // match String not ending with `?`
+    {
+      text: /\bString[^?]/g,
+      replaceValue: 'String @db.Text ',
+    },
+    // match String ending with ?
+    {
+      text: /String\?/g,
+      replaceValue: 'String? @db.Text',
+    },
+  ]
+
+  return replaceTextConfigList.reduce((fileContent, replaceConfig) => {
+    return fileContent.replace(replaceConfig.text, replaceConfig.replaceValue)
+  }, file.content)
```

Comment:
```suggestion
  return parameters.datasourceProvider !== 'mysql'
    ? file.content
    : file.content.replace(/( String\?? )/g, `$1@db.Text`)
```

Reply:
Tests will be soon finished. 

This suggested change doesn't actually work. I can check what is wrong since it is definitely simplifying my implementation.